Doctrine/RefreshTokenManager.php

```diff
@@ -23,45 +23,25 @@ class RefreshTokenManager extends BaseRefreshTokenManager
 
     protected $repository;
 
-    /**
-     * Constructor.
-     *
-     * @param ObjectManager $om
-     * @param string        $class
-     */
-    public function __construct(ObjectManager $om, $class)
+    public function __construct(ObjectManager $om, string $class)
     {
         $this->objectManager = $om;
         $this->repository = $om->getRepository($class);
         $metadata = $om->getClassMetadata($class);
         $this->class = $metadata->getName();
     }
 
-    /**
-     * @param string $refreshToken
-     *
-     * @return RefreshTokenInterface
-     */
-    public function get($refreshToken)
+    public function get(string $refreshToken): ? RefreshTokenInterface
     {
-        return $this->repository->findOneBy(array('refreshToken' => $refreshToken));
+        return $this->repository->findOneBy(['refreshToken' => $refreshToken]);
     }
 
-    /**
-     * @param string $username
-     *
-     * @return RefreshTokenInterface
-     */
-    public function getLastFromUsername($username)
+    public function getLastFromUsername(string $username): ? RefreshTokenInterface
     {
-        return $this->repository->findOneBy(array('username' => $username), array('valid' => 'DESC'));
+        return $this->repository->findOneBy(['username' => $username]);
```

Comment:
you will have more than 1 token inside database so 'valid' => 'DESC' here was taking the last token